src/components/icon/index.tsx

```diff
@@ -18,6 +21,14 @@ export type IconProps = Omit<ImageProps, 'source'> &
      * the asset group, default is "icons"
      */
     assetGroup?: string;
+    /**
+     * Badge props passed down to Badge component
+     */
+    badgeProps?: BadgeProps;
+    /**
+     * Hide badge shown on button
+     */
+    hideBadge?: boolean;
```

Comment:
I don't see why this prop is needed... Render badge only if there is a badgeProps prop